### PR TITLE
Fix/refactor faulty append structured data method cu 8693h26qh

### PR DESCRIPTION
**Pull request by annalinneajohansson**

### CODE REVIEW

File: library/Controller/ContentType/School.php [hunk 1]
```diff
@@ -46,22 +52,82 @@ public function addSecondaryContentType(ContentTypeComponentInterface $contentTy
...
+        ->image(get_the_post_thumbnail_url($postId))
+        ->url(get_permalink($postId))
+        ->numberOfStudents(WP::getField('numberOfStudents', $postId))
+        ->openingHours(WP::getField('openingHours', $postId));
+
+        if (!empty($this->getSecondaryContentType())) {
+            foreach ($this->getSecondaryContentType() as $contentType) {
+                switch ($contentType->getKey()) {
+                    case 'place':
+                        $visitingAddresses = WP::getField('visiting_address', $postId);
```
Comment (sebastianthulin) [inline library/Controller/ContentType/School.php:87]: Can we really assume that this is an array, with the keys below? May return false if field is undefined in db.

Comment (sebastianthulin) [reply library/Controller/ContentType/School.php:87]: We may set a default, array for the source by using array_merge

$visitingAdress = array_merge($default = [
'adress' => ['name' => null]
], (array) $visitingAdress);

File: library/Helper/Data.php [hunk 2]
```diff
@@ -9,46 +9,26 @@
...
+        return json_encode($structuredData, JSON_PRETTY_PRINT | JSON_UNESCAPED_UNICODE);
+    }
+    /**
+     * Alias for 'normalizeStructuredData' for legacy purposes.
+     *
+     * @param array $structuredData The structured data to be prepared.
+     * @return array The prepared structured data.
+     */
+    public function prepareStructuredData(array $structuredData = []): ?string
+    {
```
Comment (sebastianthulin) [inline library/Helper/Data.php:30]: We may use _doing_it_wrong function here, to make developers avoid the function in the future.

**Review by sebastianthulin — DISMISSED** [review]

File: library/Controller/ContentType/School.php (hunk 1 above)
Comment (annalinneajohansson) [reply library/Controller/ContentType/School.php:87]: True, I based it off the existing structure on school api posts (which won't be parsed here anyway). I'll return to this and do a retake on it.

**Comment by annalinneajohansson** [comment]
Task linked: [CU-8693h26qh Schemadata för "Person" får skolans titel istället för personens namn](https://app.clickup.com/t/8693h26qh)

File: library/Controller/ContentType/School.php (hunk 1 above)
Comment (annalinneajohansson) [reply library/Controller/ContentType/School.php]: @sebastianthulin See https://github.com/helsingborg-stad/Municipio/pull/841/files#diff-b4ab122d03a2c3ba408b751b9a864b393d59fa5a8067bb7d6a955bb97c1c3549R71-R191

File: library/Controller/ContentType/School.php [hunk 3]
```diff
@@ -46,22 +52,141 @@ public function addSecondaryContentType(ContentTypeComponentInterface $contentTy
...
+    }
+    /**
+     * Get the school address for a specific post.
+     *
+     * @param int $postId The ID of the post.
+     * @return array The array of school addresses.
+     */
+    protected function getSchoolAddress(int $postId): array
+    {
+        $addressesData = WP::getField('PostalAddress', $postId);
```
Comment (annalinneajohansson) [inline library/Controller/ContentType/School.php:135]: Any schema data will be required to be stored as a multidimensional array under the entity name defined on schema.org (case sensitive). This example would be something like array( 'PostalAddress' => array( 'streetAddress' => 'street address 123', 'postalCode' => '12345' ) );

@sebastianthulin

File: library/Controller/ContentType/Complex/Project.php [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            'schemaType' => 'Organisation',
+            'label'      => _x('Brand', 'Project brand, commonly "participants"', 'municipio')
+        ],
+        'department'  => [
+            'schemaType' => 'Organisation',
+            'label'      => _x('Department', 'Project department', 'municipio')
+        ],
+        ];
+
+        foreach ($this->getSecondaryContentType() as $contentType) {
```
Comment (sebastianthulin) [inline library/Controller/ContentType/Complex/Project.php:66]: May we need to check that this is loopable?

File: library/Helper/WP.php [hunk 5]
```diff
@@ -368,4 +390,43 @@ public static function getAttachmentCaption($postId = 0, string $postType = '')
...
+    {
+        global $post, $typenow, $current_screen;
+
+        // Check the global $typenow - set in admin.php
+        if ($typenow) {
+            return $typenow;
+        }
+
+        // Check the global $post variable - set when editing a post
+        if ($post && $post->post_type) {
```
Comment (sebastianthulin) [inline library/Helper/WP.php:409]: Check that this is a WP_Post object or isset before using ->post_type

File: library/Helper/WP.php [hunk 6]
```diff
@@ -368,4 +390,43 @@ public static function getAttachmentCaption($postId = 0, string $postType = '')
...
+        }
+
+        // Lastly check the post_type in the edit page's query string
+        if (isset($_REQUEST['post'])) {
+            $postId = intval($_REQUEST['post']);
+            return get_post_type($postId);
+        }
+
+        // Check the current screen object - set in screen settings
+        if ($current_screen && $current_screen->post_type) {
```
Comment (sebastianthulin) [inline library/Helper/WP.php:425]: Check ->post_type before using.

**Review by sebastianthulin — APPROVED** [review]

**Pull request merged**